src/BaseBlockTracker.ts

```diff
@@ -45,6 +45,12 @@ export abstract class BaseBlockTracker extends SafeEventEmitter {
     this._setupInternalEvents();
   }
 
+  async destroy() {
+    await this._maybeEnd();
+    this._cancelBlockResetTimeout();
```

Comment:
Should this come before `this._maybeEnd()`? `_maybeEnd()` might take some time to resolve for the `SubscribeBlockTracker`, because of the `eth_unsubscribe` call.

Reply:
Yeah, good call. Fixed in [7d956e6](https://github.com/MetaMask/eth-block-tracker/pull/106/commits/7d956e69f8a6a47bdeebcbc9a697f0dd57616a47).